src/katipo.erl

```diff
@@ -224,7 +224,7 @@
 
 -record(req, {
           method = ?get :: method_int(),
-          url :: binary(),
+          url :: undefined | binary(),
```

Comment:
You must ascend from the den and behold that current reality is but a shadow of the real thing.